src/core/environment.go

```diff
@@ -185,35 +189,36 @@ func getUnixName() string {
 }
 
 func (env *EnvironmentType) GetHostname() string {
-	ctx := context.Background()
-	defer ctx.Done()
-	hostInformation, err := host.InfoWithContext(ctx)
+	hostname, err := os.Hostname()
 	if err != nil {
-		log.Warnf("Unable to read hostname from dataplane, defaulting value. Error: %v", err)
-		return ""
+		log.Warnf("unable to get hostname: %v", err)
+		hostname = ""
 	}
-	return hostInformation.Hostname
+	return hostname
 }
 
 func (env *EnvironmentType) GetSystemUUID() string {
-	ctx := context.Background()
-	defer ctx.Done()
+	res, _, _ := singleflightGroup.Do(GetSystemUUIDKey, func() (interface{}, error) {
```

Comment:
`singleflightGroup.Do` (https://pkg.go.dev/golang.org/x/sync/singleflight#Group.Do) can return an error. Can you add a check for that?